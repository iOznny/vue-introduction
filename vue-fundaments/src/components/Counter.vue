<template>
    <div>
        <h2>{{ this.title }}</h2>
        <p>{{ counter }} <sup>2</sup> = {{ squareCounter }} </p>

        <div>
            <button @click="decrement">-1</button>
            <button @click="increment">+1</button>
        </div>
    </div>

</template>

<script>
    export default {
        name: 'Counter',
        props: {
            title: String,
            start: {
                type: Number,
                required: true,
                default: 10
            }
        },
        data: function() {
            return {
                counter: this.start
            }
        },
        methods: {
            getSquareValue() {
                // console.log('Get Square Value');
                return this.counter * this.counter
            },
            increment() {
                this.counter++;
            },
            decrement() {
                this.counter--;
            }
        },
        // Computated Properties
        computed: {
            squareCounter() {
                // console.log('C. Square Counter');
                return this.counter * this.counter
            },
            setTitle() {
                return this.title ? this.title : 'Counter';
            }
        }
    }
</script>

<style>
    button {
        background-color: #64BB87;
        border-radius: 5px;
        border: 1px solid white;
        color: white;
        cursor: pointer;        
        margin: 0 5px;
        padding: 5px 15px;
        transition: 0.3s ease-in-out;
    }

    button:hover {
        background-color: #5aa67b;
        transition: 0.3s ease-in-out;
    }
</style>