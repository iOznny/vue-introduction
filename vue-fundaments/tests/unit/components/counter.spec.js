// ShallowMount: Solo monta el componente que le indiquemos.
// Mount:  Monta todo el componente, sub-componentes y un proceso mas pesado de renderizado.
import { shallowMount, mount } from "@vue/test-utils";

// Components
import Counter from "@/components/Counter";

describe('Counter Component', () => {
    
    it('Match with the Snapshot', () => {
        const wrapper = shallowMount(Counter);
        expect(wrapper.html()).toMatchSnapshot();
    })
})