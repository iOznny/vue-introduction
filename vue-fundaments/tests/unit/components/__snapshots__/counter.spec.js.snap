// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Match with the Snapshot 1`] = `
<div>
  <h2></h2>
  <p>10 <sup>2</sup> = 100</p>
  <div><button>-1</button><button>+1</button></div>
</div>
`;
